x/concentrated-liquidity/lp.go

```diff
@@ -51,6 +51,7 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	cacheCtx, writeCacheCtx := ctx.CacheContext()
 	initialSqrtPrice := pool.GetCurrentSqrtPrice()
 	initialTick := pool.GetCurrentTick()
+	positionId := k.getNextPositionIdAndIncrement(ctx)
```

Comment:
Do we not need the state `KeyNextGlobalPositionId` any more since we have `KeyPositionId` and thus can be potentially replaced? (Im not sure myself, asking to see if there is a possibility to do so and worth investigating)

Reply:
There may be some way around this by iterating on the KeyPositionId prefixes and selecting the largest ID and incrementing by one, but it feels like a simpler approach imo to just have a state entry for it (would also ever just be a max of a single state entry). Please feel free to push back on this though as I could be missing something / misunderstanding your point!